JigsawSwap/Subviews/ValidationTextField.swift

```diff
@@ -0,0 +1,53 @@
+import UIKit
+import RxSwift
+import RxCocoa
+
+final class ValidationTextField: UITextField {
+    let errorMessage: String?
+    let defaultString: String?
+    let validationRegex: String
+
+    init(_ textFieldType: ValidationType, defaultString: String? = nil) {
+        self.errorMessage = textFieldType.errorMessage()
+        self.defaultString = defaultString
+        self.validationRegex = textFieldType.validationRegex()
+        super.init(frame: .zero)
+        self.placeholder = textFieldType.placeholder()
+        
+        translatesAutoresizingMaskIntoConstraints = false
+        addConstraints([
+            heightAnchor.constraint(equalToConstant: 50),
+            widthAnchor.constraint(equalToConstant: UIScreen.main.bounds.width * 0.5)
+        ])
+    }
+    
+    @available(*, unavailable, message: "Please use init() instead")
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func isValid() -> Bool {
+        if let text = self.text, text.isValid(with: validationRegex) {
+            dismissErrorMessage()
```

Comment:
Is valid powinno zwracac wartosc true albo false bez wplywania na stan textFielda

```
var isValid: Bool {
    return self.text.isValid(with: validationRegex) ?? false
}

fun setAs(_ valid: Bool) { (...) }

```

Reply:
I jesli masz juz wpakowane bezposredni do textFielda pattern to zmiany wizualne mozesz stosowac w wewnatrz textFielda